e2e/Dockerfile

```diff
@@ -41,29 +35,48 @@ RUN echo "deb http://ppa.launchpad.net/maarten-fonville/protobuf/ubuntu trusty m
     && sudo apt-key adv --keyserver hkp://keyserver.ubuntu.com:80 --recv-keys 4DEA8909DC6A13A3
 
 # install dependencies
-RUN sudo apt-get update && sudo apt-get install -y \
-    make graphviz gnupg protobuf-compiler unixodbc-dev \
-    xvfb libgtk2.0-0 libnotify-dev libgconf-2-4 libnss3 libxss1 libasound2 \
+RUN sudo apt-get update \
+    && sudo apt-get install -y --no-install-recommends \
+        apt-transport-https \
+        apt-utils \
+    && sudo apt-get install -y \
+        gnupg \
+        graphviz \
+        libasound2 \
+        libgconf-2-4 \
+        libgtk2.0-0 \
+        libnotify-dev \
+        libnss3 \
+        libxss1 \
+        make \
+        protobuf-compiler \
+        unixodbc-dev \
+        xvfb \
+    && sudo apt-get autoremove -yqq --purge \
+    && sudo apt-get clean \
     && sudo rm -rf /var/lib/apt/lists/*
 
 # install nvm
 RUN curl -o- https://raw.githubusercontent.com/creationix/nvm/v0.35.2/install.sh | bash
 
 # node vars
 ARG NVM_DIR=$HOME/.nvm
-ARG NODE_VERSION=v14.15.3
+ARG NODE_VERSION
 
 # install node and yarn
-RUN source $NVM_DIR/nvm.sh \
-    && nvm install $NODE_VERSION --lts=fermium \
+RUN set +x  \
+    && source $NVM_DIR/nvm.sh \
+    && set +x \
```

Comment:
nit: are both of the calls to `set +x` here intentional?

Reply:
Corrected. The second call should be "set -x" to improve verbosity.